### PR TITLE
feat(integration): bring React 18 fixtures in line with `indie-stack` updates

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Follow-up of #4060

### CODE REVIEW

File: integration/helpers/node-template/app/entry.client.tsx [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+  });
+};
+
+if (window.requestIdleCallback) {
+  window.requestIdleCallback(hydrate);
+} else {
+  // Safari doesn't support requestIdleCallback
+  // https://caniuse.com/requestidlecallback
+  window.setTimeout(hydrate, 1);
+}
```
Comment (MichaelDeBoey) [inline integration/helpers/node-template/app/entry.client.tsx:22]: This isn't really necessary for our integration tests, but I think it's a good idea to keep these consistent with our templates

File: package.json [hunk 2]
```diff
@@ -121,5 +121,8 @@
   },
   "engines": {
     "node": ">=14"
+  },
+  "devDependencies": {
+    "isbot": "^3.5.1"
```
Comment (mcansh) [inline package.json:126]: all dependencies in the workspace root are "dev dependencies" so we just put them all in "dependencies"

**Review by mcansh — APPROVED** [review]
LGTM, but I'll defer to @jacob-ebey

File: integration/helpers/node-template/app/entry.server.tsx [hunk 3]
```diff
@@ -12,13 +13,17 @@ export default function handleRequest(
   responseHeaders: Headers,
   remixContext: EntryContext
 ) {
+  let callbackName = isbot(request.headers.get("user-agent"))
```
Comment (jacob-ebey) [inline integration/helpers/node-template/app/entry.server.tsx:16]: MJ wants these to be two explicit code-paths, something like:
```js
if (isBot) {
  return renderForBot(...);
}

return renderForBrowser(...);
```

**Review by jacob-ebey — CHANGES_REQUESTED** [review]

**Pull request merged**